### PR TITLE
Improve compile times (Issue #114)

### PR DESCRIPTION
**Pull request by matt-holden**
Re: #114 

_This fix dropped my average re-build (no code changes) time from 60-90 seconds to 7-10 seconds._

### Summary

It appears to me that the largest contributor to compile times is `API.swift` and its re-generation in the "Generate Apollo GraphQL API" build phase.

The build script currently overwrites `API.swift` on every build, resulting in a new last-modified timestamp. As such, the swift compiler won't use its cached build of the file.  (You'll have to excuse me, I don't know much about the compiler, so I may be butchering terminology.)

In my estimation, this causes XCode to not only recompile `API.swift`, but also _all files that reference its contents_.

### Fix

Modify the "Generate Apollo GraphQL API" build phase to only overwrite `API.swift` if the output of `check-and-run-apollo-codegen.sh` results in a different output than the previous build.

The diff is hard to read, as it's in the pbxproj, so I've included the relevant lines below:

```sh
TEMP_FILE=$(mktemp)
[[ -f API.swift ]] || touch API.swift # ensure sure file exists

$APOLLO_FRAMEWORK_PATH/check-and-run-apollo-codegen.sh generate $(find . -name '*.graphql') --schema schema.json --output $TEMP_FILE

cmp API.swift $TEMP_FILE || cp $TEMP_FILE API.swift
```

In short, it outputs the generated API file to `TEMP_FILE`. Then, it
uses `cmp` to test if `TEMP_FILE` is identifical to the existing `API.swift`. If they are not, it overrwrites `API.swift` with the newly generated code.

### CODE REVIEW

**Comment by rnystrom** [comment]
> This fix dropped my average re-build (no code changes) time from 60-90 seconds to 7-10 seconds.

You are my hero 🌟 🎉 ❤️  🌮 🦅 🇺🇸 💥 🎉 😍  🌟

**Comment by rnystrom** [comment]
@matt-holden you might want to consider suggesting this change upstream:

- [Apollo iOS](https://github.com/apollographql/apollo-ios)
- [Their script docs](http://dev.apollodata.com/ios/installation.html#adding-build-step)

**Comment by matt-holden** [comment]
Will do!

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by matt-holden** [comment]
@rnystrom -- opened https://github.com/apollographql/apollo-ios/issues/147
